common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCache.java

```diff
@@ -49,6 +54,9 @@
 public class SharedPreferencesAccountCredentialCache extends AbstractAccountCredentialCache {
```

Comment:
>SharedPreferencesAccountCredentialCache [](http://example.com/codeflow?start=13&length=39)

have you explored SharedPreferencesAccountCredentialCacheWithMemoryCache ?

It already does in-memory caching. You can consider replacing SharedPreferencesAccountCredentialCache with SharedPreferencesAccountCredentialCacheWithMemoryCache and flight at that layer.

Reply:
That's a good idea. Just checked that class is doing what I am trying to introduce here.